var/spack/repos/builtin/packages/node-js/package.py

```diff
@@ -57,9 +60,15 @@ class NodeJs(Package):
 
     # https://github.com/nodejs/node/blob/master/BUILDING.md#unix-and-macos
     depends_on("gmake@3.81:", type="build")
+    depends_on("python@3.6:3.11", when="@19.1:", type="build")
+    depends_on("python@3.6:3.10", when="@16.11:19.0", type="build")
+    depends_on("python@3.6:3.9", when="@16.0:16.10", type="build")
+    depends_on("python@2.7,3.5:3.8", when="@15", type="build")
+    depends_on("python@2.7,3.6:3.10", when="@14.18.2:14", type="build")
+    depends_on("python@2.7,3.5:3.8", when="@13.1:14.18.1", type="build")
+    depends_on("python@2.7,3.5:3.7", when="@12:13.0", type="build")
```

Comment:
This is actually relatively easy to confirm/update. Simply checkout the git repo and run the following bash script:
```bash
for tag in $(git tag -l | sort -Vr)
do
    echo
    echo $tag
    echo
    git checkout $tag BUILDING.md && grep Python BUILDING.md && git restore --staged BUILDING.md && git restore BUILDING.md
done
```
Older versions stored this info in README.md instead of BUILDING.md, but we don't have any versions that old.